tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_GrTagsForHydro.cpp

```diff
@@ -0,0 +1,137 @@
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Tov.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/CloneUniquePtrs.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.ValenciaDivClean.Subcell.GrTagsForHydro",
+    "[Unit][Evolution]") {
+  using System = grmhd::ValenciaDivClean::System;
+  using gr_tag = typename System::spacetime_variables_tag;
+  using subcell_gr_tag = evolution::dg::subcell::Tags::Inactive<gr_tag>;
+  using subcell_faces_gr_tag = evolution::dg::subcell::Tags::OnSubcellFaces<
+      typename System::flux_spacetime_variables_tag, 3>;
+  using GrVars = typename gr_tag::type;
+  using SubcellGrVars = typename subcell_gr_tag::type;
+  using FaceGrVars = typename subcell_faces_gr_tag::type::value_type;
+
+  using Translation = domain::CoordinateMaps::TimeDependent::Translation;
+  using Identity2D = domain::CoordinateMaps::Identity<2>;
+  using Solution =
+      RelativisticEuler::Solutions::TovStar<gr::Solutions::TovSolution>;
+  const Mesh<3> subcell_mesh{5, Spectral::Basis::FiniteDifference,
+                             Spectral::Quadrature::CellCentered};
+  const double time = 0.1;
+  const auto logical_coords = logical_coordinates(subcell_mesh);
+  const Solution solution{1.28e-3, 100.0, 2.0};
+
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+  functions_of_time["Translation"] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+          0.0, std::array<DataVector, 3>{{{0.0}, {2.3}, {0.0}}}, 10.0);
+
+  auto box = db::create<db::AddSimpleTags<
+      Initialization::Tags::InitialTime, evolution::dg::subcell::Tags::Mesh<3>,
+      domain::Tags::ElementMap<3, Frame::Grid>,
+      domain::CoordinateMaps::Tags::CoordinateMap<3, Frame::Grid,
+                                                  Frame::Inertial>,
+      domain::Tags::FunctionsOfTime,
+      evolution::dg::subcell::Tags::Coordinates<3, Frame::Logical>,
+      Tags::AnalyticSolution<Solution>, subcell_gr_tag, subcell_faces_gr_tag>>(
+      time, subcell_mesh,
+      ElementMap<3, Frame::Grid>{
+          ElementId<3>{0},
+          domain::make_coordinate_map_base<Frame::Logical, Frame::Grid>(
+              domain::CoordinateMaps::Identity<3>{})},
+      domain::make_coordinate_map_base<Frame::Grid, Frame::Inertial>(
+          domain::CoordinateMaps::TimeDependent::ProductOf2Maps<Translation,
+                                                                Identity2D>(
+              Translation{"Translation"}, Identity2D{})),
+      clone_unique_ptrs(functions_of_time), logical_coords,
+      Solution{1.28e-3, 100.0, 2.0}, SubcellGrVars{},
```

Comment:
Could you pass in `solution` instead of making a new one (mostly to help avoid inconsistencies creeping into the code in the future)?

Reply:
no, the solutions aren't copyable. I tried that at first. I'll add `constexpr` variables for the arguments, though